jujugui/static/gui/src/app/models/handlers.js

```diff
@@ -110,6 +110,76 @@ YUI.add('juju-delta-handlers', function(Y) {
         result.tags = tags.join(',');
       }
       return result;
+    },
+
+    /**
+      Translates the new JujuStatus and WorkloadStatus's to the legacy Juju 1
+      values.
+
+      This logic is a JS implementation of the Juju core implementation to
+      accomplish the same task:
+      https://github.com/
+        juju/juju/blob/juju-1.26-alpha1/state/status_model.go#L272
+
+      @method translateToLegacyAgentState
+      @param {String} agentStatus JujuStatus.Current
+      @param {String} workloadStatus WorkloadStatus.Current
+      @param {String} workloadStatusMessage WorkloadStatus.Message
+    */
+    translateToLegacyAgentState: function(
+      agentStatus, workloadStatus, workloadStatusMessage) {
+      var statusMaintenance = 'maintenance';
```

Comment:
I think it would be nice to have those defined as constants and reusable elsewhere.